.NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DateParser.cs

```diff
@@ -191,18 +191,28 @@ protected DateTimeResolutionResult ParseImplicitDate(string text, DateObject ref
                 }
                 else
                 {
-                    futureDate = DateObject.MinValue.SafeCreateFromValue(year, month, day);
-                    pastDate = DateObject.MinValue.SafeCreateFromValue(year, month, day);
+                    if(!((year % 4 == 0)&&(year % 100 != 0) || (year % 400 == 0)) && (month == 2) && (day == 29))
+                    {
+                        futureDate = DateObject.MinValue.SafeCreateFromValue(year, month-1, day);
+                        pastDate = DateObject.MinValue.SafeCreateFromValue(year, month-1, day);
+                    }
+                    else
+                    {
+                        futureDate = DateObject.MinValue.SafeCreateFromValue(year, month, day);
+                        pastDate = DateObject.MinValue.SafeCreateFromValue(year, month, day);
+                    }
                     if (!hasMonth)
                     {
                         if (futureDate < referenceDate)
                         {
-                            futureDate = futureDate.AddMonths(1);
+                            if(((year % 4 == 0) && (year % 100 != 0) || (year % 400 == 0)) || ((month!=2) && (day != 29)))

```

Comment:
The same, reuse the method created.
Also I think this condition "((month!=2) && (day != 29))" is not correct.